test/test_ops.py

```diff
@@ -665,6 +665,19 @@ def test_inplace_forward_mode_AD(self, device, dtype, op):
 
         self._forward_grad_helper(device, dtype, op, self._get_safe_inplace(op.get_inplace()))
 
+    # Tests that functions that do not support autograd do not fail unexpectedly
+    @_gradcheck_ops(op_db)
+    def test_nondifferentiable(self, device, dtype, op):
+        # Test only when supports_autograd=True and dtype is float64
+        if op.supports_autograd or dtype.is_complex:
+            self.skipTest("Skipped! Autograd is supported or complex dtype.")
+        if "resize" in op.name:
+            self.skipTest("Skipped! Cannot resize variables that require grad.")
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
```

Comment:
Note for the future: we should make sample input generation lazy so not all the sample inputs have to be generated to acquire just the first one